poglink/cogs/rates.py

```diff
@@ -181,6 +187,8 @@ async def compare_and_notify_all(self):
 
             # Update last rates value for next iteration
             self.last_rates[idx] = current_rates
+            await asyncio.sleep(self.url_delay)
```

Comment:
Smaller fixed delay added for looping between each URL because Discord was rate limiting.

Reply:
This makes the integration test go super slowly because this is built in.

I think we should prioritize figuring out what exactly is the rate limit we're avoiding and make it so this delay as small as possible.

It still doesn't make sense to me that a momentary burst of a few messages is triggering a rate limit. Unless we're contributing to the [global rate limit](https://discord.com/developers/docs/topics/rate-limits) of 50/s. In that case, I think a 1 s delay should be sufficient.

Reply:
Unfortunately it looks like `discord.py` isn't actually storing the full context of the rate limited response so it's a bit hard to debug this further. [They only store and log the "time remaining" element](https://github.com/Rapptz/discord.py/blob/45d498c1b76deaf3b394d17ccf56112fa691d160/discord/http.py#L280-L299)of the response headers  ([here](https://discord.com/developers/docs/topics/rate-limits#header-format-rate-limit-header-examples) are the headers that we would care about). Even the bucket that the bot is logging in the warning is [supplied by the client](https://github.com/Rapptz/discord.py/blob/45d498c1b76deaf3b394d17ccf56112fa691d160/discord/http.py#L676-L684), not the one returned by the server.

Reply:
One thing to note: The discord API docs state that a [fixed delay should not be hard coded](https://discord.com/developers/docs/topics/rate-limits#:~:text=rate%20limits%20should%20not%20be%20hard%20coded%20into%20your%20bot/application) because the rate limits can change, but with discord.py not actually storing/logging the full set of relevant information, I don't think we can do this without hard-coding something.

Shame that discord.py is abandoned.

Reply:
Speaking of which
[nextcord](https://github.com/nextcord/nextcord)
[Pycord](https://github.com/Pycord-Development/pycord)

Reply:
Nice, add an issue for a future migration?

Reply:
#55